### PR TITLE
Add an analyze statement to trigger statistics updates

### PR DESCRIPTION
**Pull request by mfussenegger**
## Checklist

 - [ ] User relevant changes are recorded in ``CHANGES.txt``
 - [ ] Touched code is covered by tests
 - [ ] Documentation has been updated if necessary
 - [ ] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [ ] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: sql/src/main/java/io/crate/execution/ddl/TransportAnalyzeAction.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+/**
+ * TODO: This currently calls tableStatsService.updateStats on each node, which means each node by itself will
+ * generate the stats and populate it locally.
+ *
+ * A more efficient way of doing things would be to fetch the updated stats to the coordinator node and then publish
+ * the stats to all other nodes.
+ *
+ * Currently we do a sys.shards query to generate the stats, so this approach might be good enough for now?
+ */
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/execution/ddl/TransportAnalyzeAction.java]: @kovrus @seut any inputs to this?

Comment (mfussenegger) [reply sql/src/main/java/io/crate/execution/ddl/TransportAnalyzeAction.java]: I think I'll keep it as is in this draft. Once we extend the stats with more information we might want to change how it is generated as well anyway, so we can adapt this then. (E.g. maybe the TableStatsService will call analyze periodically, and analyze will do something else entirely)

Comment (seut) [reply sql/src/main/java/io/crate/execution/ddl/TransportAnalyzeAction.java]: Yes I'm fine to keep this as is for now. As you mentioned already we may change this anyway later on.

File: sql/src/main/java/io/crate/execution/ddl/TransportAnalyzeAction.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@Singleton
+public final class TransportAnalyzeAction implements NodeAction<AnalyzeRequest, AcknowledgedResponse> {
+
+    private static final String NAME = "internal:crate:sql/analyze";
+    private TransportService transportService;
+    private ClusterService clusterService;
+    private TableStatsService tableStatsService;
```
Comment (seut) [inline sql/src/main/java/io/crate/execution/ddl/TransportAnalyzeAction.java]: can't these be final?

**Pull request merged**